### ISSUE TITLE
Presigned POST not setting correct ACL after POSTing

### ISSUE THREAD

**Issue by karrui** [localstack/localstack#2407]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

I am using Node with `aws-cli` to create my S3 client:

```
const s3 = new aws.S3({
  region: awsRegion,
  endpoint: isDev ? 'http://localhost:4572/' : undefined,
})
```

I am generating a presigned POST url in my express server and sending it to the frontend:

```
s3.createPresignedPost({
  Bucket: imageS3Bucket,
  ...
  Fields: {
    acl: 'public-read',
    key: req.body.fileId,
    'Content-MD5': req.body.fileMd5Hash,
    'Content-Type': req.body.fileType,
  },
}
```
However, after POSTing to the URL and checking the ACL of the posted object, the public read ACL is not set:

```
$ awslocal s3api get-object-acl --bucket local-image-bucket --key <imageId>.jpg
{
    "Owner": {
        "DisplayName": "webfile",
        "ID": <ID>
    },
    "Grants": [
        {
            "Grantee": {
                "ID": <Some other ID>,
                "Type": "CanonicalUser"
            },
            "Permission": "FULL_CONTROL"
        }
    ]
    // Missing AllUsers READ permission as expected.
}
```

## Expected behavior

```
$ awslocal s3api get-object-acl --bucket local-image-bucket --key <imageId>.jpg
{
    "Owner": {
        "DisplayName": "webfile",
        "ID": <Some ID>
    },
    "Grants": [
        {
            "Grantee": {
                "ID": <Some ID>,
                "Type": "CanonicalUser"
            },
            "Permission": "FULL_CONTROL"
        },
        // The below group permission should exist
        {
            "Grantee": {
                "Type": "Group",
                "URI": "http://acs.amazonaws.com/groups/global/AllUsers"
            },
            "Permission": "READ"
        }
    ]
}
```

## Actual behavior

Posted above, `AllUsers`'s `READ` permission is missing

# Steps to reproduce

1. Generate a presignedUrl using aws's node sdk, setting acl to `public-read`: 
```
s3.createPresignedPost({
  Bucket: imageS3Bucket,
  ...
  Fields: {
    acl: 'public-read',
    key: req.body.fileId,
    'Content-MD5': req.body.fileMd5Hash,
    'Content-Type': req.body.fileType,
  },
}
```

2. POST to the given endpoint with uploaded file

3. `GET /<bucket-name>/<imageId>.jpg` and get 403'd due to not having public read permissions (even though it is already set in the `createPresignedPost` method)

## Command used to start LocalStack

docker-compose.yml consisting of the below code snippet is used:

```
localstack:
    image: localstack/localstack:latest
    container_name: localstack
    ports:
      - '4566-4599:4566-4599' # Localstack services
    environment:
      - SERVICES=s3
      - DATA_DIR=/tmp/localstack/data
      - AWS_SECRET_ACCESS_KEY
      - AWS_ACCESS_KEY_ID
      - AWS_REGION
      - DEBUG=1
    volumes:
      - ./init-localstack.sh:/docker-entrypoint-initaws.d/init-localstack.sh
      - './.localstack:/tmp/localstack'
      - '/var/run/docker.sock:/var/run/docker.sock'
```

`init-localstack.sh` is just creates buckets if they do not exist

```
#!/bin/bash
set -x
awslocal s3 mb s3://local-image-bucket
awslocal s3api put-bucket-acl --bucket local-image-bucket --acl public-read
awslocal s3 mb s3://local-logo-bucket
awslocal s3api put-bucket-acl --bucket local-logo-bucket --acl public-read
set +x
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

...

**Comment by karrui** [comment]
Seems like a regression. Was fixed as reported in #406 and a coworker confirmed that it 403 is not being returned when accessing presigned POSTs with public read POST policies on `localstack/localstack:0.8.0`

**Comment by robertcoopercode** [comment]
I've also run into this issue. I've gone back to version `0.8.0` as @karrui has mentioned it worked and I can confirm that that version doesn't throw `403` errors when accessing the POSTed S3 ressource.

However, I took a look at the ACL for an object uploaded through a POST to localstack and it appears that there is not public-read ACL for `groups/global/AllUsers`:

```
{
    "Owner": {
        "DisplayName": "webfile",
        "ID": "75aa57f09aa0c8caeab4f8c24e99d10f8e7faeebf76c078efc7c6caea54ba06a"
    },
    "Grants": [
        {
            "Grantee": {
                "ID": "75aa57f09aa0c8caeab4f8c24e99d10f8e7faeebf76c078efc7c6caea54ba06a",
                "Type": "CanonicalUser"
            },
            "Permission": "FULL_CONTROL"
        }
    ]
}
```

**Comment by Anonyymi** [comment]
Can confirm that this doesn't work in 0.11.3. Against real aws environment the objects get correct acl assigned, in localstack they just don't.

Wondering if I should just attempt to fix this myself and send a pull request...

**Comment by whummer** [comment]
Thanks for the update @Anonyymi 

> Wondering if I should just attempt to fix this myself and send a pull request...

Great - that would be very much appreciated! Please let us know if you need any help or guidance. Thanks!

**Comment by Anonyymi** [comment]
@whummer 
Well I tried for a hour or two. I got localstack dev env working so I was able to debug stuff, but I guess it just takes time to know a project's codebase. I thought debugging would be easy, I set logging level to DEBUG but the code never went where I thought it would.

I'll give it more tries later and report here if I get anything figured out. I was fiddling with current master branch, and services/s3/s3_listener.py,s3_starter.py files. More specifically, I was investigating these lines.

https://github.com/localstack/localstack/blob/master/localstack/services/s3/s3_listener.py#L1186
https://github.com/localstack/localstack/blob/master/localstack/services/s3/s3_starter.py#L64

Though now that I look at it, the latter in s3_starter.py seems to be some kind of 'fix routine' which is executed every time the s3 service is restarted?

**Comment by whummer** [comment]
Thanks for the update @Anonyymi @karrui . This should be fixed upstream in spulec/moto#3246 and is now available in LocalStack. Can you please give it another try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks!
